### ISSUE TITLE
Sites loading-slowly in Firefox 118 when AdGuard extension is enabled

### ISSUE THREAD

**Issue by dftf-stu** [AdguardTeam/AdguardBrowserExtension#2524]
### AdGuard Extension version

AdGuard Browser Extension v4.2.168

### Browser version

Firefox (x64), v118.0.1 + Microsoft Edge (x64), v117.0.2045.55

### OS version

Windows 10 Home 22H2 (x64) Build 19045.3448

### What filters do you have enabled?

AdGuard Base filter, AdGuard Mobile Ads filter, Filter unblocking search ads and self-promotion, Online Malicious URL Blocklist, NoCoin Filter List

### Issue Details

I've noticed in the last couple of weeks or so that sites are taking a lot-longer to load in Firefox than they used to.

So as a test: I have a bookmarks folder called "News", which opens eight (8) websites when I middle-click it.  I started a stopwatch on my phone when I middle-clicked it (which causes all of the sites to load at-once, in their own tab), and pressed "stop" when all sites had loaded (defined as "the favicon for each site has appeared in their respective tab").

In Firefox (118.0.1):
- With the AdGuard browser extension DISABLED, all sites had loaded in **9 seconds**
- With the AdGuard browser extension ENABLED, all sites had loaded in **56 seconds!**

In Microsoft Edge (117.0.2045.55):
- With the AdGuard browser extension DISABLED, all sites had loaded in **11 seconds**
- With the AdGuard browser extension ENABLED, all sites had loaded in **21 seconds**

I have tried fully uninstalling Firefox, deleting my local profile folders (from AppData\Local\Mozilla and AppData\Roaming\Mozilla), then re-importing my bookmarks back into the new profile, but it makes no-difference.

Is there any other troubleshooting I can do here, or is there some-conflict between a recent update to the AdGuard Browser Extension and the recent Firefox 118 release that is causing this issue?

### Expected Behavior

For websites to load quicker in Firefox 118, as they did until around a couple of weeks ago

### Screenshots

<details><summary>Screenshot 1:</summary>

</details>

### Additional Information

OS: Windows 10 Home (x64) 22H2 Build 19045.3448
AdGuard: Browser Extension v4.2.168 (same version in both browsers)
DNS: https://dns.adguard-dns.com/dns-query

**Comment by dftf-stu** [comment]
UPDATE: I've just tried installing a rival ad-block extension into Firefox (whose name sounds-like "blue-jock, horror-tin") and that one _has absolutely no-issues_ in my Firefox -- all eight sites loaded in 9-10 seconds! (So _as-fast as when **no extension**_ is used! 😲)

Not-sure what's suddenly causing AdGuard to be so-slow in Firefox, but hopefully they can fix it? 🙏🏻

**Comment by ks1322** [comment]
Probably related to https://github.com/AdguardTeam/AdguardBrowserExtension/issues/2508

**Comment by dftf-stu** [comment]
@ks1322: interesting, so you first-noticed the issue back in Firefox 117, and you're using a Linux-based OS.  Whereas I am using Firefox 118.0.1, and Windows 10.

I did a search on here for any existing open-tickets (e.g. for terms like "firefox", "slow", "delay", "hang", "freeze") before logging mine, and I did see yours, but as it only mentioned Yandex Maps in the title (a site I never use) I didn't think it would be relevant.

Do you use any Chromium-based browsers on your distro (e.g. Chromium, Brave, Opera, Vivaldi) and do you see the same CPU issues in them, with AdGuard added?  On my Windows 10 system, I am not getting major-delays in loading sites in Microsoft Edge, only in Firefox.

It's to the point of being unusable for me currently: if I have some websites open, and then open a few more (for example, I middle-click on a few articles on a news website) then it literally causes all my other sites to either freeze, or load slowly.  Images in articles don't load as I scroll down the pages, or YouTube videos stutter, or take 10-15 seconds if I skip-forward to resume playback.  I have to wait until all sites are fully-loaded before I can do anything in any of them. 😕

**Comment by dftf-stu** [comment]
@ks1322: within the reviews for the extension on the Firefox Add-Ons site, I found [this one](https://addons.mozilla.org/en-GB/firefox/addon/adguard-adblocker/reviews/2000664/) which says:

> _"Doesn't work anymore. Since two days keeps crashing Firefox, slows it down and raise CPU utilisation up to 100% for almost all time. I had reset and reinstalled addon but doesn't help. On MS Edge there is no problem."_

**Comment by dftf-stu** [comment]
@ks1322: I've just-tested the five most-popular ad-blocker extensions, as listed in the [Firefox Add Ons website](https://addons.mozilla.org/en-GB/firefox/search/?q=ad%20block&sort=users&type=extension):

![AdBlocker-Firefox-Testing-5Oct2023](https://github.com/AdguardTeam/AdguardBrowserExtension/assets/83965343/3758c367-1ddf-436a-a2f9-5d2037224cf8)

As you can see, something is definitely wrong at the moment with AdGuard in Firefox as it takes significantly-longer to finish loading my eight test websites than any of the other four leading-rivals (and uses the most RAM, too!)

**Comment by dftf-stu** [comment]
UPDATE: I did the same testing in Microsoft Edge and... AdGuard was also the slowest-performing for me there, too 😭

![AdBlocker-FirefoxEdge-Testing-5Oct2023](https://github.com/AdguardTeam/AdguardBrowserExtension/assets/83965343/975a2e26-5495-4358-b4b2-3777e84f3c70)

I'd be happy to help the developers with any testing -- but I'm honestly just wondering about moving to one of their rivals instead, given they are faster and generally use less RAM.  Is there any benefit to sticking with AdGuard?  Have any third-parties tested the various ad-blockers and confirmed it does the best job, for example? 🤔

**Comment by shape55** [comment]
I was just going to post a report on this same issue!  This started happening sometime within the past couple weeks, and I'm also using Firefox 118.0.1 w/ the latest AdGuard v4.2.168 add-on on macOS Catalina.  AG is causing webpages to load about 10x slower than in Safari (using AG for Safari), when Firefox webpages used to be significantly faster than Safari.

Even if I turn off every single toggle setting in every category of Firefox AdGuard so that nothing is enabled, it still prevents webpages from loading normally.  If I disable the add-on in the Add-ons Manager though, webpages then load instantaneously.  I also tried removing and re-adding the add-on, but this didn't help.  I've been using AG in Firefox for many years and never seen this happen before.

**Comment by maximtop** [comment]
Hi, thank you for opening this issue. I attempted to reproduce the issue by adding some news websites but was unable to do so.

> So as a test: I have a bookmarks folder called "News", which opens eight (8) websites when I middle-click it. I started a stopwatch on my phone when I middle-clicked it (which causes all of the sites to load at once, in their own tab), and pressed "stop" when all sites had loaded (defined as "the favicon for each site has appeared in their respective tab").

Could you please share the list of websites in your "News" folder and the settings of your AdGuard extension? You can do this via an export of your settings. If the settings contain sensitive information, you can email them to devteam@adguard.com with a reference to this issue and the browser extension you are using.

**Comment by dftf-stu** [comment]
Hi @maximtop 

- The extension I was using is called "AdGuard AdBlocker" which I installed from [here](https://addons.mozilla.org/en-GB/firefox/addon/adguard-adblocker) in the Firefox Add-Ons store

- The eight websites I visited in my test are: [Ars Technica](https://m.arstechnica.com) • [BleepingComputer](https://www.bleepingcomputer.com/news/) • [Guido Fawkes](https://order-order.com/) • [MoneySavingExpert](https://www.moneysavingexpert.com/news/) • [Spiked Online](https://www.spiked-online.com) • [Sky News UK](https://news.sky.com) • [The Register](https://www.theregister.com) • [Thurrott](https://www.thurrott.com/blog)

- Please find the exported settings attached (which I have renamed from JSON to TXT, as GitHub blocks JSON file-uploads)
[20231009_140828_adg_ext_settings_4.2.168.txt](https://github.com/AdguardTeam/AdguardBrowserExtension/files/12846456/20231009_140828_adg_ext_settings_4.2.168.txt)

- After re-installing AdGuard to test for you just now (and export the settings you requested) I opened all eight-sites in Firefox 118.0.1 and it took 59 seconds.  I then tested in the [most-popular rival](https://addons.mozilla.org/en-GB/firefox/addon/ublock-origin) to you, and it took just-over 9 seconds.  I would be surprised if your team are unable to replicate this, given in this ticket there are reports from users on both macOS and Linux who have experienced the issue in both Firefox 117 and 118 versions

**Comment by maximtop** [comment]
> Hi @maximtop
> 
> * The extension I was using is called "AdGuard AdBlocker" which I installed from [here](https://addons.mozilla.org/en-GB/firefox/addon/adguard-adblocker) in the Firefox Add-Ons store
> * The eight websites I visited in my test are: [Ars Technica](https://m.arstechnica.com) • [BleepingComputer](https://www.bleepingcomputer.com/news/) • [Guido Fawkes](https://order-order.com/) • [MoneySavingExpert](https://www.moneysavingexpert.com/news/) • [Spiked Online](https://www.spiked-online.com) • [Sky News UK](https://news.sky.com) • [The Register](https://www.theregister.com) • [Thurrott](https://www.thurrott.com/blog)
> * Please find the exported settings attached (which I have renamed from JSON to TXT, as GitHub blocks JSON file-uploads)
>   [20231009_140828_adg_ext_settings_4.2.168.txt](https://github.com/AdguardTeam/AdguardBrowserExtension/files/12846456/20231009_140828_adg_ext_settings_4.2.168.txt)
> * After re-installing AdGuard to test for you just now (and export the settings you requested) I opened all eight-sites in Firefox 118.0.1 and it took 59 seconds.  I then tested in the [most-popular rival](https://addons.mozilla.org/en-GB/firefox/addon/ublock-origin) to you, and it took just-over 9 seconds.  I would be surprised if your team are unable to replicate this, given in this ticket there are reports from users on both macOS and Linux who have experienced the issue in both Firefox 117 and 118 versions

Please, try to disable this setting 
<img width="626" alt="image" src="https://github.com/AdguardTeam/AdguardBrowserExtension/assets/17272769/a44be346-585f-4280-928d-628455ddb6ef">
and check if if it helped to reduce time

**Comment by maximtop** [comment]
> Hi @maximtop : yes, disabling that took the total-time down from 59 seconds, to just-over 21 seconds.
> 
> That is the same-speed I get in _Microsoft Edge_ to load all eight websites (though in _Edge_, I did have that setting enabled, but didn't experience the slowdown. So odd why having it enabled has only-recently started to affect just Firefox?) 🤷🏻‍♂️

We will check it. Thank you.

**Comment by dftf-stu** [comment]
Hi @ks1322 , @kodiakhub , @shape55 

Please follow these steps and see if it improves the speed loading your sites:

**(1)** Click the AdGuard icon
**(2)** Click the cog icon (⚙) to go into the settings
![AdGuard-TurnOffStatSend-01](https://github.com/AdguardTeam/AdguardBrowserExtension/assets/83965343/1d8998bb-0049-49c8-969a-dc599d3dddeb)

**(3)** Click "Additional Settings" in the left-hand list
![AdGuard-TurnOffStatSend-02](https://github.com/AdguardTeam/AdguardBrowserExtension/assets/83965343/8c2547cd-5ca2-40e7-b3a2-049ca8344d00)

**(4)** If "Help with the development of AdGuard filters" is turned on, turn it off
![AdGuard-TurnOffStatSend-03](https://github.com/AdguardTeam/AdguardBrowserExtension/assets/83965343/a6eab64f-b745-4574-a1cf-b904cd9ff6bb)

**(5)** Fully close Firefox (all open tabs and windows), wait 10-15 seconds, then load-up Firefox again.
Then try going to your usual sites and see if they load faster

**Comment by ks1322** [comment]
I can confirm that disabling "Send anonymous statistics of ad filters usage that will help us improve and optimize them" option improves Firefox performance in my test on https://yandex.ru/maps/.

**Comment by dftf-stu** [comment]
(@maximtop: this is unrelated, but seeing as you've replied here I thought I'd ask: where are we supposed to go to report issues with websites when the AdGuard DNS service is used?  There isn't a GitHub here for it, and the online report tool doesn't have an option in the dropdown for the DNS product.  I e-mailed your team, but last I heard-anything was three-weeks ago.  Is there a site we're supposed to use for this?  Thanks)

**Comment by maximtop** [comment]
> (@maximtop: this is unrelated, but seeing as you've replied here I thought I'd ask: where are we supposed to go to report issues with websites when the AdGuard DNS service is used? There isn't a GitHub here for it, and the online report tool doesn't have an option in the dropdown for the DNS product. I e-mailed your team, but last I heard-anything was three-weeks ago. Is there a site we're supposed to use for this? Thanks)

If it's a filtering issue, it's better to report it from the product where the AdGuard DNS service is incorporated. Alternatively, you can manually create an issue here: https://github.com/AdguardTeam/AdguardFilters/issues

**Comment by dftf-stu** [comment]
@maximtop: I couldn't find this repository when I first looked (before I asked you) but would this be the correct place?
[https://github.com/AdguardTeam/AdGuardDNS](https://github.com/AdguardTeam/AdGuardDNS) ?

I have a website which is totally-broken when AdGuard DNS is used, but is fine when I use any-other DNS service

**Comment by maximtop** [comment]
> @maximtop: I couldn't find this repository when I first looked (before I asked you) but would this be the correct place? https://github.com/AdguardTeam/AdGuardDNS ?
> 
> I have a website which is totally-broken when AdGuard DNS is used, but is fine when I use any-other DNS service

yes, you can report technical issues there

**Comment by dftf-stu** [comment]
@maximtop: thanks, I have reported it there as issue 676.  I'd suggest deleting this comment, and my previous two (along with your replies) to keep posts in this topic related to the original issue

**Comment by dftf-stu** [comment]
@kodiakhub : I did think your issue may-not be related, yes.

Is there any-reason on that website why you need to scroll up-and-down the page a lot? 🤔

If you do that in any browser, I would expect increased CPU and GPU use.

Have you tried disabling AdGuard and seeing if the issue still occurs then?  (You can disable it by going to the hamburger-menu (three-lines) then "Add ons and themes..." and then use the ON/OFF toggle to turn it off.  This will keep all of your settings for AdGuard so you can turn it back-on later and not lose anything.)

![AdGuard-Disabled-Firefox](https://github.com/AdguardTeam/AdguardBrowserExtension/assets/83965343/e3f28903-7e14-441d-9d28-605f423b940f)

**Comment by shape55** [comment]
@dftf-stu: Ok, yes, disabling this single setting AND relaunching Firefox definitely made sites load faster.  If I don't relaunch the browser after disabling this though, then it makes no difference, even if I wait a while.

However, this is only part of the issue. Just having the add-on enabled still makes pages load much more slowly than they did a couple weeks ago, w/ all toggles in every category turned off.  I also just updated to FF 118.0.2, but this had no effect.  However, if it requires relaunching the browser after disabling other settings for changes to take effect, then I wouldn't be able to tell.

**Comment by dftf-stu** [comment]
@shape55: I'm not a _macOS_ user, only _Windows_, so I cannot confirm first-hand what options _AdGuard for Safari_ offers -- but the screenshots [here](https://adguard.com/kb/adguard-for-safari/overview) for that version don't seem to show it.  Though I'm not-sure if _Safari_ is even affected by this slowdown issue anyway.  On my PCs, only _Firefox_ was, but _Chrome_ and _Edge_ weren't

As for the slowdown in _Firefox_ -- the person who works for _AdGuard_ here, @maximtop , has said they are looking into it.  All I can suggest in the meantime is to disable your _AdGuard_ extension and use a different ad-blocker, such as [this](https://addons.mozilla.org/en-GB/firefox/addon/ublock-origin/) one, until they update this ticket to say the slowdown issue has been fixed.  Then switch-back to _AdGuard_ at that-time if you wish

**Comment by shape55** [comment]
@dftf-stu: Yeah, I just realized that the Safari version doesn't include it, and then changed my question shortly before you replied :)

For now, I'll just keep this one setting disabled, as it's not worth the time to switch to a new ad blocker unless there is no fix for a while.  But thanks for the suggestion.

**Comment by dftf-stu** [comment]
Hi @maximtop 

Just to let you know but I've ran the test on a different PC while visiting family and the results are:

**Firefox**
- "Help with the development of AdGuard filters" ON  = 1 min 40 sec
- "Help with the development of AdGuard filters" OFF = 42 sec

**Microsoft Edge**
- "Help with the development of AdGuard filters" ON  = 51 sec
- "Help with the development of AdGuard filters" OFF = 38 sec

So having that option on definitely makes a noticeable difference to the load-time of sites, but more-so in Firefox, where it added nearly a minute extra, compared to the 13 seconds in Edge

**Comment by dftf-stu** [comment]
Hi @maximtop 

I've just tested on another laptop I have (which has an Intel Core i7, 16GB RAM and an SSD) and the results are:

**Firefox v118.0.2**
- "Help with the development of AdGuard filters"  ON = 29 sec
- "Help with the development of AdGuard filters" OFF = 11 sec
- Using the most-popular rival ad-blocker extension    =   5 sec

**Microsoft Edge v118.0.2088.46**
- "Help with the development of AdGuard filters" ON = 11 sec
- "Help with the development of AdGuard filters" OFF =   9 sec
- Using the most-popular rival ad-blocker extension    = 11 sec

So AdGuard was actually faster than the most-popular ad-blocker extension for me in Edge when the report-back option is turned-off, and the same-speed as it when it is left on.  However, in Firefox, [your rival](https://addons.mozilla.org/en-GB/firefox/addon/ublock-origin/) loaded all eight websites in 5 seconds, whereas AdGuard with the report-back option disabled took 11 seconds, and 29 SECONDS when it is enabled.  That is nearly three-times as long (or nearly SIX TIMES as long as your biggest rival)

Not-sure what is going on with AdGuard and Firefox lately, but hopefully it can be fixed... any updates you could share?

**Comment by dftf-stu** [comment]
Hi @maximtop 

Have there been any updates on this issue you could share?  Your last reply was three weeks ago (9 Oct), and this ticket will have been open for a month as of this Thursday.

I've already advised any friend or family-member to whom I've previously recommended your extension to disable that setting, but within the next week or two I'll simply start recommending they remove AdGuard and switch to a different ad-blocker as I cannot be-sure if this issue is still being looked-into or not.

Thanks

**Comment by maximtop** [comment]
Hi @dftf-stu,

I apologize for the delay in updating you on this matter. We have been preoccupied with other pressing issues, which has resulted in a delay in investigating the page loading times you've reported.

We are still committed to resolving this issue, and your information has been valuable in helping us understand the scope. That said, it may take some time before we can provide a concrete update or solution.

Thank you for your patience.

**Comment by shape55** [comment]
Hi, just wanted to check back on this to see what the status was.  Have any changes been made w/ the `Help with the development of AdGuard filters` setting, or general site loading times as reported above?

**Comment by shape55** [comment]
Hi, has there been any update on this since Oct 30, 2023?

**Comment by dftf-stu** [comment]
Hi @shape55 

Yes, it is surprising that @maximtop has no updates since their last-reply, which is now just-over 15 months ago.

I've just done a quick re-test: AdGuard's speed has improved, but is still slower, and it uses more RAM, than its main-rival.

Using the latest 64-bit Firefox for Windows (v134.0.2) I tested opening eight sites at the same-time with uBO, and then with AdGuard (with only one of those add-ons active at one time, and all other add-ons disabled), fully-closing Firefox between each test.  Once the favicon on each tab had appeared, I stopped the timer.

And the results were:
- [uBO](https://addons.mozilla.org/en-US/firefox/addon/ublock-origin/) (v1.62.0) loaded all eight sites in around 9-10 seconds (and Firefox peaked at 1045 MB of RAM, down to 840 MB idle)
- AdGuard (v4.4.49) with "Help with the development of AdGuard filters" off took 16-17 seconds (1485 MB -> 1125 MB RAM)
- AdGuard (v4.4.49) with "Help with the development of AdGuard filters" on took 27-28 seconds (1575 MB -> 1295 MB RAM)

Compared to the 56 seconds it took when I first raised this issue, 16-17 seconds is clearly an improvement.  (Though AdGuard still uses around ~450-530 MB more RAM at peak, then ~275-450 MB more RAM once all sites were loaded).

As none of the AdGuard development team have offered any updates on this issue in the 15 months since the "thank you for your patience" message above, I'd say it suggests desktop Firefox is not a priority for them.

Perhaps consider looking into [uBO](https://addons.mozilla.org/en-US/firefox/addon/ublock-origin/) instead, unless you specifically need to stick with AdGuard?

**Comment by shape55** [comment]
@dftf-stu: Thank you very much for providing this detailed and helpful update and comparison.

Actually, I used to use uBlock Origin for macOS many years ago, and then switched to AdGuard b/c uBO took far too much time to manage at the time – but from the new screenshots it looks like they have made it more user-friendly.  I love AG’s simplicity, interfaces/design, and customization, but just wish that it was as fast as it was before whatever caused this issue in early Oct. 2023.

@kodiakhub: This isn’t a feature request though, and neither of us are complaining.  It’s a bug that’s been needing to be resolved for nearly 1.5 years now which prevents important feedback from being submitted to AdGuard (due to needing to keep [this setting](https://github.com/AdguardTeam/AdguardBrowserExtension/issues/2524#issuecomment-1753006199) disabled), so that they can improve their filters.  You were in agreement with us in your comments back in Oct. 2023, as it was causing major issues for you w/ Firefox:

https://github.com/AdguardTeam/AdguardBrowserExtension/issues/2524#issuecomment-1750269256
https://github.com/AdguardTeam/AdguardBrowserExtension/issues/2524#issuecomment-1753328071
https://github.com/AdguardTeam/AdguardBrowserExtension/issues/2524#issuecomment-1753382629

**Comment by Mr-McMuffin** [comment]
The issue is still present, Adguard in 2025 even with the signature is unreliable or nearly unuseable. 
Needs a major attention to detail, super slow exactly as the OP mentioned.

**Comment by maximtop** [comment]
Hi everyone,

Thank you for your patience regarding this issue. For some time, our team has been focused on migrating to Manifest V3, which has been a major undertaking and led to delays in addressing some important issues, including this one.

We still have a few high-priority tasks to work through, but we plan to revisit this issue and explore possible improvements to memory usage as soon as possible.

We truly appreciate your feedback and support!

**Comment by nvimtor** [comment]
Only thing holding me back from purchasing a license; it is unbelievably slow, with all websites. Hope this gets fixed soon.

**Comment by maximtop** [comment]
Hi @dftf-stu (and others),

I attempted to reproduce your performance measurement method (opening 8 bookmarks and timing until favicons appear). It seems this method might be sensitive to factors beyond raw loading speed. For instance, I observed that tabs sometimes don't visually update to the "loaded" state until interacted with (like hovering the mouse), which could affect stopwatch timing. I recorded a brief screencast showing this: https://uploads.adguard.org/screencast_2025-04-07_17-35-07.mp4

To get more consistent technical metrics, I tried a different approach using Firefox's developer tools. I tested one site from your list, `https://arstechnica.com/`. In the DevTools Network tab, I disabled the cache, simulated a 'Regular 4G' connection (https://uploads.adguard.org/Image_2025-04-07_18-02-35.png), and reloaded the page five times under each condition:
1.  uBlock Origin (uBO) with default settings.
2.  AdGuard configured with additional filters enabled (as defaults didn't block same requests as uBO on this site).
3.  No ad blocker active.

Here are the average results based on the initial performance data provided:

| Metric           | uBO (Default) | AdGuard (v4.4.49)* | No Blocker |
| :--------------- | :------------ | :----------------- | :--------- |
| Requests         | ~48.4         | ~40.4              | ~211.2     |
| DOMContentLoaded | ~1.83 s       | ~1.83 s            | ~1.40 s    |
| Load Time        | ~2.50 s       | ~1.85 s            | ~6.82 s    |

In this specific, single-page test data, the load times for AdGuard and uBO were almost the same, and both were better than loading the page without any blocker.

And just to confirm, all this testing was done with the **"Help with the development of AdGuard filters" (statistics collection) setting disabled.** As discussed before, enabling this feature is known to significantly impact performance, especially in Firefox, because it needs to analyze page content and style changes to gather data. We recommend keeping this setting off for the best performance.

If you have another specific, repeatable test case that clearly shows the slowdown, please share the details, and I'll gladly investigate further when possible.

**Comment by dftf-stu** [comment]
Hi @maximtop 

I've just re-tested for you using the following setup:
Windows 10 22H2 (x64) • Firefox (x64) v137.0 • uBO (Manifest v2), v1.63.2 • AdGuard (Mv2), v4.4.49

I first set my homepage and new tab page in Firefox to be a blank page.

Then after, I did the following steps each test:
1. Launch Firefox
2. Middle-click on a "News" folder, which launches 7 websites, each into their own tab within the same window
3. Begin stopwatch on my phone, and stop it once all favicons had appeared
4. Disable or enable the relevant add-ons for the next test
5. Close Firefox fully, and wait for "firefox.exe" to disappear fully from Task Manager
 
Note: when Firefox is closed, cookies are not cleared, but cache is

Test results:

**AdGuard**
– with "Help with the development of AdGuard Filters" **ON**: 14 seconds
– with "Help with the development of AdGuard Filters" **OFF**: 13 seconds

**uBO**
– using my current settings: 9 seconds

So... compared to when I first logged this issue, it does appear for me the issue is resolved.

However, I did log it back in October 2023, back on Firefox v118.0.1.
We're now just-over 1 year and 6 months on, and I'm using Firefox v137.0.

So... problem I have is: were this slowdown to occur again in future, is there much point reporting it then if it
may take as long again to look into?  I'd imagine whatever the cause was, it's just resolved-itself... 🤔🤷🏻‍♂️
